File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/events/inheritance/presenter/InheritanceMainPresenter.java
package com.nerdscorner.mvplib.testapp.events.inheritance.presenter;

import com.nerdscorner.mvplib.events.presenter.BaseActivityPresenter;
import com.nerdscorner.mvplib.testapp.events.inheritance.model.InheritanceMainModel;
import com.nerdscorner.mvplib.testapp.events.inheritance.model.InheritanceMainModel.BackgroundTaskCompletedEvent;
import com.nerdscorner.mvplib.testapp.events.inheritance.view.InheritanceMainView;
import com.nerdscorner.mvplib.testapp.events.inheritance.view.InheritanceMainView.ActionClickedEvent;

import org.greenrobot.eventbus.Subscribe;

public class InheritanceMainPresenter extends BaseActivityPresenter<InheritanceMainView, InheritanceMainModel> {

    public InheritanceMainPresenter(InheritanceMainView view, InheritanceMainModel model) {
        super(view, model);
    }

    @Subscribe
    public void onActionClicked(ActionClickedEvent event) {
        model.doSomethingInBackground();
    }

    @Subscribe
    public void onBackgroundTaskCompleted(BackgroundTaskCompletedEvent event) {
        view.setTextValue("Background task completed");
    }
}

File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/events/inheritance/presenter/InheritanceMainPresenter.kt
package com.nerdscorner.mvplib.testapp.events.inheritance.presenter

import com.nerdscorner.mvplib.events.presenter.BaseActivityPresenter
import com.nerdscorner.mvplib.testapp.events.inheritance.model.InheritanceMainModel
import com.nerdscorner.mvplib.testapp.events.inheritance.model.InheritanceMainModel.BackgroundTaskCompletedEvent
import com.nerdscorner.mvplib.testapp.events.inheritance.view.InheritanceMainView
import com.nerdscorner.mvplib.testapp.events.inheritance.view.InheritanceMainView.ActionClickedEvent
import org.greenrobot.eventbus.Subscribe

class InheritanceMainPresenter(view: InheritanceMainView, model: InheritanceMainModel) :
        BaseActivityPresenter<InheritanceMainView, InheritanceMainModel>(view, model) {

    @Subscribe
    fun onActionClicked(event: ActionClickedEvent) {
        view.setTextValue("Executing background task...")
        model.doSomethingInBackground()
    }

    @Subscribe
    fun onBackgroundTaskCompleted(event: BackgroundTaskCompletedEvent) {
        view.setTextValue("Background task completed")
    }
}
